clickhouse-spark-32-integration/src/test/scala/xenon/clickhouse/single/ClickHouseDataTypeSuite.scala

```diff
@@ -14,15 +14,46 @@
 
 package xenon.clickhouse.single
 
+import org.apache.spark.sql.types._
+import org.apache.spark.sql.QueryTest.checkAnswer
+import org.apache.spark.sql.Row
 import xenon.clickhouse.{BaseSparkSuite, Logging}
 import xenon.clickhouse.base.ClickHouseSingleMixIn
 
 class ClickHouseDataTypeSuite extends BaseSparkSuite
     with ClickHouseSingleMixIn
     with SparkClickHouseSingleMixin
+    with SparkClickHouseSingleTestHelper
     with Logging {
 
-  test("write supported data types") {}
+  test("write supported data types") {
+    val structFields =
+      StructField("id", DataTypes.LongType, false, Metadata.fromJson("""{"comment": "sort key"}""")) ::
+        StructField("col_string", DataTypes.StringType, false) ::
+        StructField("col_array_string", DataTypes.createArrayType(DataTypes.StringType, false), false) :: Nil
+    val db = "t_w_s_db"
+    val tbl = "t_w_s_tbl"
+    withTable(db, tbl, structFields) {
+      val tblSchema = spark.table(s"$db.$tbl").schema
+      // TODO v2 create table should respect element nullable of array field
```

Comment:
Not sure if it is a spark bug or mine, further investigation is needed.

Reply:
It should not be considered as bug, but a limitation of SQL syntax.

For instance,
```sql
creata table xxx (abc  ARRAY<STRING> NOT NULL)
```
There is not way to mark the inner type `STRING` as `NULL` or `NOT NULL`, then it will be treat as `NULL` by default, and be converted to ClickHouse type `Array<Nullable>` eventually.